src/libp2p/Libp2p.Protocols.Noise/NoiseProtocol.cs

```diff
@@ -60,6 +60,15 @@ public async Task DialAsync(IChannel downChannel, IChannelFactory? upChannelFact
         NoiseHandshakePayload? msg1Decoded = NoiseHandshakePayload.Parser.ParseFrom(buffer.AsSpan(0, msg1.BytesRead));
         PublicKey? msg1KeyDecoded = PublicKey.Parser.ParseFrom(msg1Decoded.IdentityKey);
         //var key = new byte[] { 0x1 }.Concat(clientStatic.PublicKey).ToArray();
+        if (_extensions.StreamMuxers.Any())
+        {
+            var selectedProtocol = upChannelFactory?.SubProtocols.FirstOrDefault(proto => proto.Id == _extensions.StreamMuxers[0]);
+            context.SpecificProtocolRequest = new ChannelRequest
```

Comment:
upChannelFactory will have multistreamselect protocol only on top. But you anyway do not need it: on dial async you need yo analyze `msg1Decoded` that came from remote peer and see what muxers it supports, then select common one.
ListenAsync contains msg2Decoded which can be used for the same.